controllers/workspace_controller_runs.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	tfc "github.com/hashicorp/go-tfe"
+	appv1alpha2 "github.com/hashicorp/terraform-cloud-operator/api/v1alpha2"
+)
+
+func (r *WorkspaceReconciler) reconcileRuns(ctx context.Context, w *workspaceInstance, workspace *tfc.Workspace) error {
+	w.log.Info("Reconcile Runs", "msg", "new reconciliation event")
+
+	if workspace.CurrentRun == nil {
+		w.log.Info("Reconcile Runs", "msg", "there is no current run")
+		return nil
+	}
+
+	if w.instance.Status.Run == nil {
+		w.instance.Status.Run = &appv1alpha2.RunStatus{}
+	}
+
+	// Update current run status
+	if workspace.CurrentRun.ID != w.instance.Status.Run.ID || !w.instance.Status.Run.RunCompleted() {
```

Comment:
Should the above initialisation of `w.instance.Status.Run` happen inside this `if` as well?

What happens if we end up `setting w.instance.Status.Run` to empty struct, but then this condition doesn't evaluate and we skip the code inside it and just return with an empty (but not nil) `w.instance.Status.Run`?

Reply:
Made some changes as we discussed. Thank you!